src/main/java/com/conveyal/taui/controllers/TimetableController.java

```diff
@@ -0,0 +1,71 @@
+package com.conveyal.taui.controllers;
+
+import com.conveyal.taui.models.AddTripPattern;
+import com.conveyal.taui.models.Modification;
+import com.conveyal.taui.models.Project;
+import com.conveyal.taui.models.Region;
+import com.conveyal.taui.persistence.Persistence;
+import com.conveyal.taui.util.JsonUtil;
+import com.mongodb.QueryBuilder;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.Request;
+import spark.Response;
+
+import java.util.Collection;
+
+import static spark.Spark.get;
+
+/**
+ * Created by evan siroky on 5/3/18.
+ */
+public class TimetableController {
+    private static final Logger LOG = LoggerFactory.getLogger(TimetableController.class);
+
+    public static String getTimetables (Request req, Response res) {
+        JSONArray json = new JSONArray();
+        Collection<Region> regions = Persistence.regions.findAllForRequest(req);
+
+        for (Region region : regions) {
```

Comment:
Let's change this and only return Modifications, Projects, or Regions if they have Timetables that can be selected. No need to show an entire region of projects and modifications if there are no timetables available. The selection code could be simplified on the front end also if that assumption can be made.